packages/api/cms-api/src/auth/guards/global.guard.ts

```diff
@@ -1,60 +1,64 @@
-import { CanActivate, ExecutionContext, HttpException, Injectable } from "@nestjs/common";
+import { CanActivate, ExecutionContext, HttpException, mixin } from "@nestjs/common";
 import { Reflector } from "@nestjs/core";
 import { GqlExecutionContext } from "@nestjs/graphql";
-import { AuthGuard } from "@nestjs/passport";
+import { AuthGuard, IAuthGuard, Type } from "@nestjs/passport";
 import { Request } from "express";
 import { isObservable, lastValueFrom } from "rxjs";
 
+import { CurrentUserInterface } from "../current-user/current-user";
 import { allowForRoleMetadataKey } from "../decorators/allow-for-role.decorator";
-import { CurrentUser } from "../dto/current-user";
 
-@Injectable()
-export class GlobalAuthGuard extends AuthGuard() implements CanActivate {
-    constructor(private reflector: Reflector) {
-        super();
-    }
-
-    getRequest(context: ExecutionContext): Request {
-        return context.getType().toString() === "graphql"
-            ? GqlExecutionContext.create(context).getContext().req
-            : context.switchToHttp().getRequest();
-    }
-
-    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any
-    handleRequest<CurrentUser>(err: unknown, user: any): CurrentUser {
-        if (err) {
-            throw err;
+function CometAuthGuard(type?: string | string[]): Type<IAuthGuard> {
+    class GlobalAuthGuard extends AuthGuard(type) implements CanActivate {
+        constructor(private reflector: Reflector) {
+            super();
         }
-        if (user) {
-            return user;
+
+        getRequest(context: ExecutionContext): Request {
+            return context.getType().toString() === "graphql"
+                ? GqlExecutionContext.create(context).getContext().req
+                : context.switchToHttp().getRequest();
         }
-        throw new HttpException("AccessTokenInvalid", 200);
-    }
 
-    async canActivate(context: ExecutionContext): Promise<boolean> {
-        if (this.reflector.getAllAndOverride("disableGlobalGuard", [context.getHandler(), context.getClass()])) {
-            return true;
+        // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any
+        handleRequest<CurrentUserInterface>(err: unknown, user: any): CurrentUserInterface {
+            if (err) {
+                throw err;
+            }
+            if (user) {
+                return user;
+            }
+            throw new HttpException("UserNotAuthenticated", 200);
```

Comment:
We should research how authentication errors are handled in GraphQL. I can't believe that throwing an exception with status 200 is the best solution. But this is something for another PR